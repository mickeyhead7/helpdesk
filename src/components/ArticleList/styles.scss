.articleList {
  $desktopUp: "only screen and (min-width: 60.0625em)";

  margin: 0 -1em 2em;

  @media #{$desktopUp} {
    display: flex;
    flex-wrap: wrap;
  }

  .article {
    margin: 0 1em 2em;

    @media #{$desktopUp} {
      display: flex;
      flex-direction: column;
      width: calc(50% - 2em);
    }

    .articleContent {
      max-height: 15em;
      min-height: 5em;
      overflow: hidden;
      position: relative;

      &::after {
        background-image: linear-gradient(transparent, #fff);
        bottom: 0;
        content: "";
        display: block;
        height: 5em;
        left: 0;
        position: absolute;
        width: 100%;
      }

      p {
        margin: 0;
      }
    }
  }
}
