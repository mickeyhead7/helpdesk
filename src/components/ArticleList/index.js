import React from 'react';
import Link from 'gatsby-link';
import PropTypes from 'prop-types';
import slugify from 'slugify';
import Card from '@packages/patterns/Card';

import './styles.scss';

const ArticleList = ({ articles }) => (
  <section className="articleList">
    {articles.map((article) => {
      const { node: { frontmatter } } = article;
      const { excerpt, path, title } = frontmatter;

      return (
        <article className="articlePartial" key={slugify(path)}>
          <Card>
            <header>
              <h3>
                <Link to={path}>{title}</Link>
              </h3>
            </header>
            <div className="articleContent">
              <p>{excerpt}</p>
            </div>
          </Card>
        </article>
      );
    })}
  </section>
);

ArticleList.propTypes = {
  articles: PropTypes.arrayOf(PropTypes.shape({
    node: PropTypes.shape({
      frontmatter: PropTypes.shape({
        excerpt: PropTypes.string.isRequired,
        path: PropTypes.string.isRequired,
        title: PropTypes.string.isRequired,
      }).isRequired,
    }).isRequired,
  })).isRequired,
};

export default ArticleList;
