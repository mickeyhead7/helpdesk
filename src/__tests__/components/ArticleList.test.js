import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import ArticleList from '../../Components/ArticleList';

const articles = [
  {
    node: {
      frontmatter: {
        excerpt: 'This is an excerpt',
        path: '/article-1',
        title: 'Article one',
      },
    },
  },
  {
    node: {
      frontmatter: {
        excerpt: 'This is an excerpt',
        path: '/article-2',
        title: 'Article two',
      },
    },
  },
];

test('ArticleList', () => {
  const component = shallow(<ArticleList articles={articles} />);

  expect(toJson(component)).toMatchSnapshot();
});
