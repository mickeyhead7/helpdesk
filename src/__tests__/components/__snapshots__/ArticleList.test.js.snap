// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArticleList 1`] = `
<section
  className="articleList"
>
  <article
    className="articlePartial"
    key="article-1"
  >
    <Card>
      <header>
        <h3>
          <Link
            to="/article-1"
          >
            Article one
          </Link>
        </h3>
      </header>
      <div
        className="articleContent"
      >
        <p>
          This is an excerpt
        </p>
      </div>
    </Card>
  </article>
  <article
    className="articlePartial"
    key="article-2"
  >
    <Card>
      <header>
        <h3>
          <Link
            to="/article-2"
          >
            Article two
          </Link>
        </h3>
      </header>
      <div
        className="articleContent"
      >
        <p>
          This is an excerpt
        </p>
      </div>
    </Card>
  </article>
</section>
`;
